### DISCUSSION TITLE
自定义组件包含arrayTable,有快捷的办法获取到schema定义的table columns吗

### DISCUSSION THREAD

**Discussion by movefreechen** [alibaba/formily#2886] [Q&A 2.x] [answered]
自定义组件包含arrayTable,有快捷的办法获取到schema定义的table columns吗

**Comment by janryWang** [comment, marked as answer]
useFieldSchema，自己解析出来

**Reply by movefreechen** [reply]
你好，请问我 给 ArrayTable 传入columns参数尝试动态改变columns 发现没有效果，也尝试Observer模式，请问有什么办法能实现需求

**Reply by movefreechen** [reply]
动态改变column目的是 表格有一栏需要通过Jsx实现一些复杂的交互操作 我这边是希望通过schema渲染出表格 动态插入这一栏

**Reply by janryWang** [reply]
ArrayTable不支持传columns，自己写个自定义组件吧
